### DISCUSSION TITLE
visualize ressource characteristics in the Squest instance

### DISCUSSION THREAD

**Discussion by kogarastu** [HewlettPackard/squest#676] [General]
Hello,

At first thank you for this product.
I'm working for a big french Insurance company as Infrastructure Automation engineer. 
We working actually on a project in which we want to replace an internal marketplace Portail (internal devel) :)
it'a a big word for just talking about a Portal for ordering On premise IaaS (Vmware). We looking for a solution that is able to manage ours VMs lifecycle.
We have already an ansible AWX workflow & Playbooks for building this IaaS.

As Squest has the capability to manage resources lifecycle through the instance, (create, modify, delete)
Is there a way to modify an squest instance for example with a specific API REST to add VM technicals charactéristics  and visualise theses in the instance view ?:

- Auto generated VM name for Squest Instance name (In the playbook, we using a specific incremental system based on a API REST compliant reférential)
- DNS entries (A,cname)
- vCPUs Number
- Memory size
- Disks names
- disks sizes
- VM comments
- Maybe others informations...

If there's no way to do that,i ask myself the following question :
How can we modify vCPUs number or memory size without knowing these Characteristics in the Squest instance?

Best Regards,

Koga

**Comment by a-belhadj** [comment]
Hello,
Get a look at that [this documentation](https://hewlettpackard.github.io/squest/1.10.5/manual/service_catalog/concept/)

In an Instance you can store your data as a JSON, so you can store something like that:
```json,
    "vm_uuid": "5005769b-5569-decf-982a-ddc172f9b99b",
    "vm_name": "vm-001",
    "vcenter_hostname": "myvcenter.local",
    "vcenter_cluster": "my_cluster",
    "vcenter_folder": "my_folder",
    "vcenter_network": "VM Network",
    "vcenter_datastore": "my_datastore",
    "vcenter_datacenter": "my_datacenter",
    "vm_vcpu": 12,
    "vm_memory_gb": 32,
    "vm_disk_list": [
        {
            "size_gb": 1000,
            "type": "thin",
            "datastore": "my_datastore"
        }
    ],
 ```
 This JSON is sent to AWX as extra_vars each time you ask for an operation.
 
 Squest will do the job for your use case !
 
 Anthony

**Reply by kogarastu** [reply]
Thanks you, it's great. i have found the swagger.
i'll study it and i'll give a feedback.

**Comment by Sispheor** [comment]
You need to read the doc: https://hewlettpackard.github.io/squest/1.10.6/manual/service_catalog/concept/
